File: Data Structures & Algorithms/top-k-elements-in-list/submission-7.py
import heapq
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        c = {}
        for n in nums:
            c[n] = 1+c.get(n, 0)
        heap = []
        for key in c.keys():
            heapq.heappush(heap,(c[key], key))
            if len(heap) > k:
                heapq.heappop(heap)
        res = []
        for i in range(k):
            res.append(heapq.heappop(heap)[1])
        return res

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-7.cpp
class Solution {
public:
    int characterReplacement(std::string s, int k) {
        int l = 0, res = 0, mf = 1;
        unordered_map<char, int> mp;
        mp[s[l]]++;
        for (int r = 1; r < s.size(); r++) {
            mp[s[r]]++;
            mf = max(mf, mp[s[r]]);
            while(r-l+1-k>mf) {
                mp[s[l]]--;
                l++;
            }
            res = max(res, r-l+1);
        }
        return res;
    }
};